synfig-studio/src/gui/canvasview.cpp

```diff
@@ -207,9 +208,14 @@ class studio::CanvasViewUIInterface : public UIInterface
 		);
 
 		dialog.set_secondary_text(details);
-		dialog.add_button(button1, RESPONSE_NO);
+		Gtk::Button* close_button = dialog.add_button(button1, RESPONSE_NO);
 		dialog.add_button(button2, RESPONSE_CANCEL);
 		dialog.add_button(button3, RESPONSE_YES);
+		//add action buttons if closed without saving
+		if (button1.compare(_("Close without Saving")) == 0)
```

Comment:
Any particular reason for using `compare()` instead of `==` directly?

Reply:
No reason. It seems the == is calling .compare under the hood, but it is easier to read so I changed it to use == directly.